File: archiV1java/product/src/main/java/om/product/dao/ProductRepository.java
package om.product.dao;

import om.product.entity.Product;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.Collection;

@Repository
public interface ProductRepository extends MongoRepository<Product,String> {
    Collection<Object> findOneByName();
}

File: snapshotV2kotlin/product/src/main/kotlin/om/product/repo/ProductRepository.kt
package om.product.repo

import om.product.entity.Product
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface ProductRepository : MongoRepository<Product,String> {
    fun findBySkuCodeIgnoreCase(skuCode: String?): List<Product?>?
    fun findByDescIgnoreCase(desc: String?): List<Product?>?
    fun findBySkuCodeOrderBySkuCodeAsc(skuCode: String?): List<Product?>?
    fun findBySkuCodeOrderBySkuCodeDesc(skuCode: String?): List<Product?>?
}